crates/types/protobuf/restate/cluster.proto

```diff
@@ -76,3 +76,39 @@ message PartitionProcessorStatus {
   // Set if replay_status is CATCHING_UP
   optional restate.common.Lsn target_tail_lsn = 11;
 }
+
+enum NodeSetSelectionStrategyKind {
+  NodeSetSelectionStrategyKind_UNKNOWN = 0;
+  StrictFaultTolerantGreedy = 1;
+}
+
+message NodeSetSelectionStrategy { NodeSetSelectionStrategyKind kind = 1; }
+
+message ReplicatedProviderConfig {
+  string replication_property = 1;
+  NodeSetSelectionStrategy nodeset_selection_strategy = 2;
+}
+
+message DefaultProvider {
+  string provider = 1;
```

Comment:
It's mainly for compatibility yes, since `provider` is just a name. Plus it's the same type we are using in the `seal-and-extend-chain` request. It is also easier to convert to a ProviderKind type than a numeric protobuf enum.